Source/Request.swift

```diff
@@ -414,3 +323,283 @@ extension Request: CustomDebugStringConvertible {
         return components.joined(separator: " \\\n\t")
     }
 }
+
+// MARK: -
+
+/// Specific type of `Request` that manages an underlying `URLSessionDataTask`.
+open class DataRequest: Request {
+
+    // MARK: Helper Types
+
+    enum Requestable: TaskConvertible {
+        case request(URLRequest)
+
+        func task(session: URLSession, adapter: RequestAdapter?, queue: DispatchQueue) -> URLSessionTask {
+            var task: URLSessionTask!
+
+            switch self {
+            case let .request(urlRequest):
+                let urlRequest = urlRequest.adapt(using: adapter)
+                queue.sync { task = session.dataTask(with: urlRequest) }
```

Comment:
Not necessarily for this PR, but to avoid things like this in the future, it might be nice to have a convenience value synchronization wrapper in the future:
```swift
extension DispatchQueue {
    
    func syncAround<T>(_ closure: () -> T) -> T {
        var out: T!
        sync { out = closure() }
        return out
    }
    
}
```

Just an example, but would avoid the IUO dance here and could be returned directly if the `switch` was a `guard`.